src/components/query-list/query-list.js

```diff
@@ -0,0 +1,110 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames/bind';
+
+import styles from './query-list.css';
+import ScrollView from '../scroll-view';
+import Impagination from '../impagination';
+
+const cx = classnames.bind(styles);
+
+export default class QueryList extends Component {
+  static propTypes = {
+    type: PropTypes.string.isRequired,
+    page: PropTypes.number,
+    pageSize: PropTypes.number,
+    loadHorizon: PropTypes.number,
+    itemHeight: PropTypes.number,
+    notFoundMessage: PropTypes.string,
+    collection: PropTypes.object.isRequired,
+    fetch: PropTypes.func.isRequired,
+    renderItem: PropTypes.func.isRequired,
+    onPage: PropTypes.func
+  };
+
+  static defaultProps = {
+    page: 1,
+    pageSize: 25,
+    notFoundMessage: 'Not Found'
+  }
+
+  state = {
+    readOffset: (this.props.page - 1) * this.props.pageSize
+  };
+
+  handleUpdate = (readOffset) => {
+    let { pageSize, onPage } = this.props;
+    let page = Math.floor(readOffset / pageSize) + 1;
```

Comment:
If we track the offset in the url, and not the page, then we don't have to do this math.

Reply:
I'm all for tracking the offset in the url and not the page, but would we have to change it back to page if other people think having `page` in the URL is more useful than `readOffset`? Maybe we should keep this for now and open an issue where other Ebsco folks could chime in.

Reply:
The only reason they might prefer `page` in the url is the comfort that comes with familiarity. In terms of linkability though, `offset` is an order of magnitude more precise because if we're using page as our cursor we can be off by as several viewports worth of records depending on your page size. That's a degraded UX.

My take: we're already demonstrating a better way, and providing a much enhanced experience, so we should push forward with it.